scim/client.bal

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# This client is used to connect to the SCIM2 APIs of Asgardeo for Inbound User Provisioning.
+@display {label: "SCIM Client Connector"}
+public client class Client {
+    final http:Client clientEndpoint;
+
+    # Initializes the SCIM client.
+    #
+    # + connectorConfig - The connector configuration
+    public isolated function init(ConnectorConfig connectorConfig) returns error? {
+        http:OAuth2ClientCredentialsGrantConfig config = {
+            tokenUrl: string `${HOST_URL}/${TENANT_PATH}/${connectorConfig.orgName}/oauth2/token`,
+            clientId: connectorConfig.clientId,
+            clientSecret: connectorConfig.clientSecret,
+            scopes: connectorConfig.scope
+        };
+        self.clientEndpoint = check new (string `${HOST_URL}/${TENANT_PATH}/${connectorConfig.orgName}/${SCIM}`, {auth: config, httpVersion: "1.1"});
+    }
+
+    # Gets the list of users.
+    #
+    # + return - The list of users
+    @display {label: "Get Users"}
+    public isolated function getUsers() returns UserResponse|error {
+        UserResponse response = check self.clientEndpoint->get(USERS);
+        return response;
+    }
+
+    # Gets a user by the user ID.
+    #
+    # + id - The ID of the user
+    # + return - The user
+    @display {label: "Get User by ID"}
+    public isolated function getUser(@display {label: "User Id"} string id) returns UserResource|error {
```

Comment:
There might have requirements to get specific attributes of the user. Shall we give the support for attributes and excludedAttributes query params?
Same applies for get a group by id endpoint as well